### PR TITLE
NOISSUE - Implement manager client heartbeat mechanism 

### PR DESCRIPTION
**Pull request by SammyOina**
Introduced a heartbeat system in the manager service to maintain an active link with client agents. The updates involve sending periodic heartbeat signals controlled via the new environment variable `MANAGER_HEARTBEAT_INTERVAL`, defaulting to 1 second. The protobuf definitions, service interfaces, and server logic have been updated accordingly to handle the incoming signals. This change allows better tracking of active client connections and could be used for features like auto-reconnect or resource cleanup for lost connections in the future.

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -107,6 +110,16 @@ func main() {
 		logger.Error(err.Error())
 		return
 	}
+	hc, err := managerClient.Heartbeat(ctx)
+	if err != nil {
+		logger.Error(err.Error())
+		return
+	}
+	hbs := heartbeat.New(hc, cfg.HeartbeatInterval)
```
Comment (drasko) [inline cmd/manager/main.go:118]: Why `hbs` and not `hb`?

File: manager/api/grpc/server.go [hunk 2]
```diff
@@ -60,3 +64,16 @@ func (s *grpcServer) Process(stream manager.ManagerService_ProcessServer) error
...
 	return eg.Wait()
 }
+
+func (s *grpcServer) Heartbeat(stream manager.ManagerService_HeartbeatServer) error {
+	p, ok := peer.FromContext(stream.Context())
+	if !ok {
+		return errors.New("failed to get peer from context")
+	}
+	for {
+		if _, err := stream.Recv(); err != nil {
```
Comment (drasko) [inline manager/api/grpc/server.go:74]: What is this that we are receiving? I thought that in the `Heartbeat ()` fnc we have just to send a heartbeat message.

**Review by drasko — CHANGES_REQUESTED** [review]

File: manager/api/grpc/server.go (hunk 2 above)
Comment (SammyOina) [reply manager/api/grpc/server.go:74]: this on the server side where heartbeat requests are received

**Comment by drasko** [comment]
Closed in the favor of https://github.com/ultravioletrs/cocos/issues/104

**Pull request closed without merging**